### ISSUE TITLE
net-p2p/ethminer requires >=app-eselect/eselect-opencl-1.1.0-r4

### ISSUE THREAD

**Issue by EoD** [reagentoo/gentoo-overlay#8]
If you are using ``dev-libs/ocl-icd`` as the primary OpenCL provider, you need ``>app-eselect/eselect-opencl-1.1.0-r1`` (and I tested it successfully with ``>=app-eselect/eselect-opencl-1.1.0-r4``).

As the previous version of eselect-opencl only provide OpenCL 1.1 headers and hence compilation fails with a lot of weird errors.

This applies already to ``net-p2p/ethminer-0.12.0``.

**Comment by reagentoo** [comment]
I have successfuly build with only `CL_VERSION_1_2 - found` (eselect opencl mesa).

**Comment by EoD** [comment]
If you are using ``eselect opencl mesa``, then you are using the mesa headers and not the "global" headers (as they are called in ``app-eselect/eselect-opencl-1.1.0-r1``).

If you are using ``eselect opencl ocl-icd``, build failures will occur.

See:
https://bugs.gentoo.org/557138
https://bugs.gentoo.org/567534
https://bugs.gentoo.org/521734

**Comment by reagentoo** [comment]
But it is possible to build with `eselect-opencl-1.1.0-r1` and `mesa` eselected. Maybe should to do eselect management from ebuild and more warnings instead of adding `>app-eselect/eselect-opencl-1.1.0-r1` to `RDEPEND`.

**Comment by reagentoo** [comment]
Also builds with `eselect opencl beignet` and `eselect-opencl-1.1.0-r1`:
```
-- Looking for CL_VERSION_2_0 - found
```

**Comment by EoD** [comment]
Yeah, as I mentioned above the problematic "global" headers are those provided by the old version of eselect-opencl and those are only used if you select ocl-icd. All other cases probably do work as eselect-opencl will then symlink the system headers to the vendor-specific headers (see ``ls -al /usr/include/CL/``).
